src/streamlink/plugins/picarto.py

```diff
@@ -11,9 +11,15 @@
 
 @pluginmatcher(re.compile(r"""
     https?://(?:www\.)?picarto\.tv/
-    (?:(?P<po>streampopout|videopopout)/)?
-    (?P<user>[^&?/]+)
-    (?:\?tab=videos&id=(?P<vod_id>\d+))?
+    (?:
+        (?:(?P<po_live>streampopout)/(?P<po_user>[^/]+)/public)
+    |
+        (?:(?P<po_vod>videopopout)/(?P<po_vod_id>\d+))
+    |
+        (?:(?:[^/]+)/videos/(?P<vod_id>\d+))
+    |
+        (?P<user>[^/?&]+)
+    )$
```

Comment:
The three wrapping non-capture groups are redundant and the `po_live` and `po_vod` ones are not needed, as you can already get the match type (live/vod) from the `po_user` and `po_vod_id` capture groups.

Reply:
Ah, good point, cheers.